File: rules/llm02-sensitive-info/go/hardcoded-api-key-in-client.go
package main

import (
	"os"

	"github.com/anthropics/anthropic-sdk-go"
	anthropicoption "github.com/anthropics/anthropic-sdk-go/option"
	"github.com/openai/openai-go"
	"github.com/openai/openai-go/option"
)

func badClients() {
	// ruleid: hardcoded-api-key-in-llm-client-go
	openaiClient := openai.NewClient(option.WithAPIKey("sk-fake-key-for-testing"))
	_ = openaiClient

	// ruleid: hardcoded-api-key-in-llm-client-go
	claude := anthropic.NewClient(anthropicoption.WithAPIKey("sk-ant-fake-key"))
	_ = claude
}

func goodClients() {
	// ok: hardcoded-api-key-in-llm-client-go
	openaiClient := openai.NewClient(option.WithAPIKey(os.Getenv("OPENAI_API_KEY")))
	_ = openaiClient

	// ok: hardcoded-api-key-in-llm-client-go
	claude := anthropic.NewClient()
	_ = claude
}

File: rules/llm02-sensitive-info/typescript/hardcoded-api-key-in-client.ts
import Anthropic from "@anthropic-ai/sdk";
import { createAnthropic } from "@ai-sdk/anthropic";
import { createOpenAI } from "@ai-sdk/openai";
import { GoogleGenerativeAI } from "@google/generative-ai";
import OpenAI from "openai";

// ruleid: hardcoded-api-key-in-llm-client-js
const openai = new OpenAI({ apiKey: "sk-fake-key-for-testing" });

// ruleid: hardcoded-api-key-in-llm-client-js
const claude = new Anthropic({ apiKey: "sk-ant-fake-key", maxRetries: 2 });

// ruleid: hardcoded-api-key-in-llm-client-js
const vercelOpenAI = createOpenAI({ apiKey: "sk-fake-key" });

// ruleid: hardcoded-api-key-in-llm-client-js
const vercelClaude = createAnthropic({ apiKey: "sk-ant-fake" });

// ruleid: hardcoded-api-key-in-llm-client-js
const gemini = new GoogleGenerativeAI("AIzaFakeKeyForTesting");

// ok: hardcoded-api-key-in-llm-client-js
const goodOpenAI = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

// ok: hardcoded-api-key-in-llm-client-js
const goodClaude = new Anthropic();

// ok: hardcoded-api-key-in-llm-client-js
const goodVercel = createOpenAI({ apiKey: process.env.OPENAI_API_KEY! });
